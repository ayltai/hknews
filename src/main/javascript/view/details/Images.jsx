import PropTypes from 'prop-types';
import React from 'react';
import { ArrayField, ImageField, } from 'react-admin';

import { Gallery, } from '../list/Gallery';

export const Images = props => {
    if (props && props.record && props.record.images && props.record.images.length > 0) return (
        <ArrayField
            {...props}
            label=''
            source='images'>
            <Gallery
                caption='description'
                source='url'>
                <ImageField
                    label=''
                    source='url' />
            </Gallery>
        </ArrayField>
    );

    return <span />;
};

Images.propTypes = {
    record : PropTypes.object,
};
