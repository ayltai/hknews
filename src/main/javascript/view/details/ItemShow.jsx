import { makeStyles, } from '@material-ui/core';
import React from 'react';
import { ArrayField, DateField, ImageField, RichTextField, SimpleShowLayout, ShowController, ShowView, TextField, } from 'react-admin';

import { Gallery, } from '../list/Gallery';
import { ScrollToTopOnMount, } from '../ScrollToTopOnMount';

export const ItemShow = props => {
    const classes = makeStyles({
        title : {
            paddingBottom : '0.75em',
            fontSize      : '120%',
            fontWeight    : 'bold',
        },
    })();

    return (
        <React.Fragment>
            <ScrollToTopOnMount />
            <ShowController {...props}>
                {controllerProps => (
                    <ShowView
                        {...props}
                        {...controllerProps}>
                        <SimpleShowLayout>
                            <RichTextField
                                className={classes.title}
                                label=''
                                source='title' />
                            <RichTextField
                                label=''
                                source='description' />
                            {controllerProps && controllerProps.record && controllerProps.record.images && controllerProps.record.images.length > 0 && (
                                <ArrayField
                                    label=''
                                    source='images'>
                                    <Gallery
                                        label='description'
                                        source='imageUrl'>
                                        <ImageField
                                            label=''
                                            source='imageUrl' />
                                    </Gallery>
                                </ArrayField>
                            )}
                            <TextField
                                label=''
                                source='sourceName' />
                            <DateField
                                label=''
                                source='publishDate'
                                showTime />
                        </SimpleShowLayout>
                    </ShowView>
                )}
            </ShowController>
        </React.Fragment>
    );
};
