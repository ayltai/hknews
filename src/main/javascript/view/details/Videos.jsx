import { makeStyles, } from '@material-ui/core';
import PropTypes from 'prop-types';
import React from 'react';
import { BigPlayButton, ControlBar, LoadingSpinner, Player, } from 'video-react';
import 'video-react/dist/video-react.css';

import { Constants, } from '../../Constants';

export const Videos = props => {
    const classes = makeStyles(theme => ({
        container : {
            display : 'flex',
        },
        video     : {
            margin : theme.spacing(1),
        },
    }))();

    if (props && props.record && props.record.videos && props.record.videos.length > 0) return (
        <div className={classes.container}>
            {props.record.videos.map(video => video.url.startsWith('https://www.youtube.com/') || video.url.startsWith('https://youtu.be/') ? (
                <iframe
                    key={video.id}
                    className={classes.video}
                    title={video.url}
                    frameBorder={0}
                    allowFullScreen={true}
                    width={Constants.VIDEO_SIZE}
                    height={Constants.VIDEO_SIZE / 16 * 9}
                    src={video.url} />
            ) : (
                <Player
                    key={video.id}
                    className={classes.video}
                    fluid={false}
                    width={Constants.VIDEO_SIZE}
                    poster={video.cover}
                    src={video.url}>
                    <LoadingSpinner />
                    <BigPlayButton position='center' />
                    <ControlBar autoHide={false} />
                </Player>
            ))}
        </div>
    );

    return <span />;
};

Videos.propTypes = {
    record : PropTypes.object,
};
