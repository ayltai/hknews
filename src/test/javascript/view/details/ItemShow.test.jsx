import { mount, } from 'enzyme';
import React from 'react';
import { TestContext, } from 'react-admin';
import { act, } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';

import { ItemShow, } from '../../../../main/javascript/view/details/ItemShow';

describe('<ItemShow />', () => {
    const component = (
        <TestContext initialState={{
            admin : {
                customQueries : {},
            }
        }}>
            <ItemShow
                basePath='/'
                resource='foo' />
        </TestContext>
    );

    it('renders without errors', () => {
        expect(renderer.create(component).toJSON()).toMatchSnapshot();
    });

    it('mounts without errors', async () => {
        await act(async () => {
            mount(component);
        });
    });
});
