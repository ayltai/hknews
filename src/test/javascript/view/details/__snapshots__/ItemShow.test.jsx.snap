// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemShow /> renders without errors 1`] = `
<div
  className="show-page RaShow-root-2"
  translate={[Function]}
>
  <div
    className="RaShow-main-3 RaShow-noActions-4"
  >
    <div
      className="MuiPaper-root MuiCard-root RaShow-card-5 MuiPaper-elevation1 MuiPaper-rounded"
    />
  </div>
</div>
`;
